File: python/openral_py/model/identity.py


from typing import Any, Dict, Optional


class Identity:
    def __init__(
        self, 
        uid: str, 
        name: Optional[str] = None, 
        site_tag: Optional[str] = None, 
        alternateIDs: list = [], 
        alternateNames: list = []
    ):
        self.uid = uid
        self.name = name
        self.site_tag = site_tag
        self.alternate_ids = alternateIDs
        self.alternate_names = alternateNames

    def to_map(self):
        return {
            "UID": self.uid,
            "name": self.name,
            "siteTag": self.site_tag,
            "alternateIDs": self.alternate_ids,
            "alternateNames": self.alternate_names,
        }
    
    @staticmethod
    def from_map(data: Dict[str, Any]):
        return Identity(
            uid=data["UID"],
            name=data.get("name"),
            site_tag=data.get("siteTag"),
            alternateIDs=data.get("alternateIDs", []),
            alternateNames=data.get("alternateNames", []),
        )

File: flutter/openral_flutter/lib/model/identity.dart
class Identity {
  final String uid;
  final String? name;
  final String? siteTag;
  final List<String> alternateIDs;
  final List<String> alternateNames;

  Identity({
    required this.uid,
    this.name,
    this.siteTag,
    this.alternateIDs = const [],
    this.alternateNames = const [],
  });

  Map<String, dynamic> toMap() {
    return {
      "UID": uid,
      "name": name,
      "siteTag": siteTag,
      "alternateIDs": alternateIDs,
      "alternateNames": alternateNames,
    };
  }
}
